cmd/flexvolume/cmd/mount.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2017 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/rook/rook/pkg/agent/flexvolume"
+	"github.com/spf13/cobra"
+)
+
+var (
+	mountCmd = &cobra.Command{
+		Use:   "mount",
+		Short: "Mounts the volume to the pod volume",
+		RunE:  mount,
+	}
+)
+
+func init() {
+	RootCmd.AddCommand(mountCmd)
+}
+
+func mount(cmd *cobra.Command, args []string) error {
+
+	client, err := getRPClient()
+	if err != nil {
+		return fmt.Errorf("Rook: Error getting RPC client: %v", err)
+	}
+
+	var opts = &flexvolume.AttachOptions{}
+	if err := json.Unmarshal([]byte(args[1]), opts); err != nil {
+		return fmt.Errorf("Rook: Could not parse options for mounting %s. Got %v", args[1], err)
+	}
+	opts.MountDir = args[0]
+
+	log(client, fmt.Sprintf("calling agent to attach volume %s/%s", opts.Pool, opts.Image), false)
+	var devicePath string
+	err = client.Call("FlexvolumeController.Attach", opts, &devicePath)
+	if err != nil {
+		return fmt.Errorf("Rook: Mount volume failed: %v", err)
```

Comment:
errors on Attach are not getting logged in the agent pod logs...consider logging the error here (with stdout redirect) so that the agent pod log will also have it.

or just log it from inside the Attach() function, before it returns the error back to the flex driver.

Reply:
Correct. This would not be logged in the agent. I thought it was not necessary since the error is sent to kubernetes who will then make it available via the `kubectl describe pod` command. But i think it is worth it to log it in the agent as well.  I will do that